rest/server_context.go

```diff
@@ -595,13 +595,19 @@ func (sc *ServerContext) getOrAddDatabaseFromConfig(ctx context.Context, config
 func GetBucketSpec(ctx context.Context, config *DatabaseConfig, serverConfig *StartupConfig) (base.BucketSpec, error) {
 
 	var server string
-	if config.Server != nil {
+	if config.Server != nil && *config.Server != "" {
```

Comment:
I've moved the check for empty string from `_getOrAddDatabaseFromConfig`, i'm doing the check inside the `GetBucketSpec`